src/Controls/src/Core/Internals/PropertyPropagationExtensions.cs

```diff
@@ -26,10 +28,12 @@ internal static void PropagatePropertyChanged(string propertyName, Element eleme
 			if (propertyName == null || propertyName == Shell.NavBarIsVisibleProperty.PropertyName)
 				BaseShellItem.PropagateFromParent(Shell.NavBarIsVisibleProperty, element);
 
-			foreach (var child in children)
+			foreach (var child in children.ToList())
```

Comment:
I would say it's better to use `ToArray` if it's just readonly operation, since the compiler will do some optimizations for arrays.


```suggestion
			foreach (var child in children.ToArray())
```

Reply:
Good idea!